core/keyboard_nav/navigation.js

```diff
@@ -569,58 +545,36 @@ Blockly.navigation.disconnectBlocks_ = function() {
   Blockly.getMainWorkspace().getCursor().setCurNode(connectionNode);
 };
 
-/*************************/
-/** Keyboard Navigation **/
-/*************************/
+/** ***************** */
+/** Helper Functions  */
+/** ***************** */
 
 /**
- * Finds where the cursor should go on the workspace. This is either the top
- * block or a set position on the workspace.
+ * Move the marker to the cursor's current location.
  * @private
  */
-Blockly.navigation.focusWorkspace_ = function() {
-  Blockly.hideChaff();
-  var cursor = Blockly.getMainWorkspace().getCursor();
-  var reset = Blockly.getMainWorkspace().getToolbox() ? true : false;
-  var topBlocks = Blockly.getMainWorkspace().getTopBlocks(true);
-
-  Blockly.navigation.resetFlyout_(reset);
-  Blockly.navigation.currentState_ = Blockly.navigation.STATE_WS;
-  if (topBlocks.length > 0) {
-    cursor.setCurNode(Blockly.navigation.getTopNode(topBlocks[0]));
-  } else {
-    var ws = Blockly.getMainWorkspace();
-    // TODO: Find the center of the visible workspace.
-    var wsCoord = new Blockly.utils.Coordinate(100, 100);
-    var wsNode = Blockly.ASTNode.createWorkspaceNode(ws, wsCoord);
-    cursor.setCurNode(wsNode);
-  }
+Blockly.navigation.markAtCursor_ = function() {
+  Blockly.getMainWorkspace().getMarker().setCurNode(
```

Comment:
There a couple other instances where you could make use of local variable to store Workspace, but here particularly, it would make it easier to read

Reply:
done